bcda/web/api.go

```diff
@@ -179,6 +179,17 @@ func bulkRequest(resourceTypes []string, w http.ResponseWriter, r *http.Request,
 		return
 	}
 
+	// After successfully creating the job, we need to mark the job as FAILED if we had any other issues
+	// with populating data associated with the job.
+	defer func() {
+		if err != nil {
+			if err1 := db.Model(&newJob).Update("status", "Failed").Error; err1 != nil {
```

Comment:
Should we retry here?  A job in `pending` state blocks ACOs from making any requests (resulting in `429` responses).
Also, does it make sense to create a ticket to monitor for jobs in `Pending` state?  I think this would be good insight... it could warn us of thinks like: (1) workers not keeping up with jobs in the queue, (2) jobs stuck in `Pending`, etc

Reply:
Ah totally forgot how bad this would be if the clean up fails (ACO will be blocked indefinitely). I was thinking it's a minor inconvenience and the caller can retry. 

Since it's that critical that we have a fully formed job (and that has several error scenarios like we noted). What are your thought about putting the job creation into a transaction? We can pre-compute quite a bit of the information before opening the transaction. And then we have guarantees that all jobs that are in the pending state are valid. That way we don't get into the situation where an ACO is blocked indefinitely.

Reply:
> Also, does it make sense to create a ticket to monitor for jobs in `Pending` state? I think this would be good insight... it could warn us of thinks like: (1) workers not keeping up with jobs in the queue, (2) jobs stuck in `Pending`, etc

Would our current JobQueueCount alert be sufficient to cover these scenarios? Assuming that all `Pending` jobs are valid.

Reply:
Transaction makes sense to me, if you think its feasible/in-scope for this ticket.  And yes, good point about `JobQueueCount` alert... that should be sufficient here!